### DISCUSSION TITLE
Notification created by

### DISCUSSION THREAD

**Discussion by scottolsen** [excid3/noticed#154] [Ideas]
Curious if you've ever thought about adding who created the notification to the table? Like a `created_by_id` or `creator_id`. I know we can pass that in to the params currently but seemed like something that could be useful as an actual attribute.

**Comment by excid3** [comment]
Hmm, good question. I could see that being useful. Are there any benefits from not just writing that to the params though?

**Reply by mechiland** [reply]
One common scenario is that creating a notification list screen like this:

<img width="570" alt="image" src="https://github.com/excid3/noticed/assets/33785/892629da-7d43-43b9-b13f-45838bed0ce4">

To display the creator's avatar and name, setting `creator` (`actor` might be a better name) in params is doable but it will trigger second query to get the user. Also, it's kind of common use case, each notification should belong to an actor. Have a foreign key constraint in the `notifications` table will be better for data integrity other than setting arbitory value in a json payload.  :-)

**Reply by janpeterka** [reply]
I really like this, that would enable me to have "activity feed" quite easily.

**Comment by scottolsen** [comment]
I'm not sure if you can have the association work out of the box like it would adding it to the table. Similar to how you currently can do `notification.recipient` you could do `notification.creator`.

**Comment by andrezimpel** [comment]
+1
